perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1368,6 +1368,23 @@ public int getIdOfBlockedLogin(PerunSession sess, String login, String namespace
 		}
 	}
 
+	@Override
+	public Integer getRelatedUserIdByBlockedLoginInNamespace(PerunSession sess, String login, String namespace) throws LoginIsNotBlockedException {
+		try {
+			int userId;
+			if (namespace == null) {
+				userId =  jdbc.queryForInt("SELECT related_user_id FROM blocked_logins WHERE login=? AND namespace IS NULL", login);
+			} else {
+				userId = jdbc.queryForInt("SELECT related_user_id FROM blocked_logins WHERE login=? AND namespace=?", login, namespace);
+			}
+			return userId == 0 ? null : userId;
```

Comment:
is this check necessary?

Reply:
This was a nitpick to begin with, but the behavior you're looking for is the 'queryForObject()' method from jdbc. 'queryForInt' was implemented to supplement a jdbc method that has been deprecated (and all it does is add the conversion from null to 0, that you revert here :D ) 

![Screenshot from 2023-04-03 10-13-53](https://user-images.githubusercontent.com/97669007/229452168-d78242bd-2232-4ca6-baac-1f98dcdc1b16.png)